src/containers/PDisk/PDisk.tsx

```diff
@@ -46,20 +50,37 @@ export function PDisk() {
     }, [dispatch]);
 
     const fetchData = useCallback(
-        (isBackground: boolean) => {
+        (isBackground?: boolean) => {
             if (!isBackground) {
                 dispatch(setPDiskDataWasNotLoaded());
             }
+
             if (nodeId && pDiskId) {
-                dispatch(getPDiskData({nodeId, pDiskId}));
-                dispatch(getPDiskStorage({nodeId, pDiskId}));
+                return Promise.all([
+                    dispatch(getPDiskData({nodeId, pDiskId})),
+                    dispatch(getPDiskStorage({nodeId, pDiskId})),
+                ]);
             }
+
+            return undefined;
         },
         [dispatch, nodeId, pDiskId],
     );
 
     useAutofetcher(fetchData, [fetchData], true);
 
+    const handleRestart = () => {
```

Comment:
```suggestion
    const handleRestart = async () => {
```
and let's not change type `ButtonWithConfirmDialogProps`